calico_rkt/calico_rkt.py

```diff
@@ -1,138 +1,149 @@
 
 def main():
     mode = env['CNI_COMMAND']
 
     if mode == 'init':
-        print_stderr('No initialization work to perform')
+        logger.info('No initialization work to perform')
     elif mode == 'ADD':
-        print_stderr('Executing Calico pod-creation plugin')
-        create(
-            container_id=env['CNI_CONTAINERID']
-            )
+        logger.info('Executing Calico pod-creation plugin')
+        create(container_id=env['CNI_CONTAINERID'])
     elif mode == 'DEL':
-        print_stderr('Executing Calico pod-deletion plugin')
-        delete(
-            container_id=env['CNI_CONTAINERID']
-            )
+        logger.info('Executing Calico pod-deletion plugin')
+        delete(container_id=env['CNI_CONTAINERID'])
 
 def create(container_id):
     """"Handle rkt pod-create event."""
-    print_stderr('Configuring pod %s' % container_id, file=sys.stderr)
+    logger.info('Configuring pod %s' % container_id)
     netns_path='%s/%s/%s' % (NETNS_ROOT, container_id, env['CNI_NETNS'])
     _datastore_client = IPAMClient()
 
     try:
         endpoint, ip = _create_calico_endpoint(container_id=container_id,
-                                            netns_path=netns_path,
-                                            client=_datastore_client)
+                                               netns_path=netns_path,
+                                               client=_datastore_client)
 
-        _create_profile(endpoint=endpoint, 
-                        profile_name=input_json['name'], 
+        _create_profile(endpoint=endpoint,
+                        profile_name=INPUT_JSON['name'],
                         ip=ip,
                         client=_datastore_client)
     except CalledProcessError as e:
-        print_stderr('Error code %d creating pod networking: %s\n%s' % (
+        logger.info('Error code %d creating pod networking: %s\n%s' % (
             e.returncode, e.output, e))
         sys.exit(1)
-    print_stderr('Finished Creating pod %s' % container_id)    
+    logger.info('Finished Creating pod %s' % container_id)
 
 def delete(container_id):
     """Cleanup after a pod."""
-    print_stderr('Deleting pod %s' % container_id, file=sys.stderr)
+    logger.info('Deleting pod %s' % container_id)
 
     _datastore_client = IPAMClient()
 
     # Remove the profile for the workload.
     _container_remove(hostname=HOSTNAME,
-                        orchestrator_id=ORCHESTRATOR_ID,
-                        container_id=container_id,
-                        client=_datastore_client)
-
-    # Delete profile
-    try:
-        _datastore_client.remove_profile(input_json['name'])
-    except:
-        print_stderr("Cannot remove profile %s; Profile cannot be found." % container_id)
+                      orchestrator_id=ORCHESTRATOR_ID,
+                      container_id=container_id,
+                      client=_datastore_client)
+
+    profile_name = INPUT_JSON['name']
+
+    # Delete profile if only member
```

Comment:
I think this opens a race-condition if running a Calico + rkt deployment across multiple nodes.

E.g. One node creates a container in this network, at the same time another deletes the last one.  The new container could come up, and then its profile could be deleted on the other node.

This is a known issue in profile management, and probably requires libcalico work to make it safe, but worth knowing about.